fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -2686,6 +2686,19 @@ public ChangedTransactionDetail adjustExistingTransaction(final LoanTransaction
         transactionForAdjustment.reverse(reversalExternalId);
         transactionForAdjustment.manuallyAdjustedOrReversed();
 
+        if (transactionForAdjustment.getTypeOf().equals(LoanTransactionType.MERCHANT_ISSUED_REFUND)) {
+            for (LoanTransaction loanTransaction : getLoanTransactions()) {
```

Comment:
You might wanna rewrite this foreach to use stream api and if it finds suitable related transaction, then reverse it.